src/main/scala/lc/Main.scala

```diff
@@ -64,13 +91,53 @@ class Captcha {
   } 
 }
 
+case class Size(height: Int, width: Int)
+case class Parameters(level: String, media: String, input_type: String, size: Option[Size])
+case class Id(id: String)
+case class Answer(answer: String, id: String)
+
 object LCFramework{
   def main(args: scala.Array[String]) {
+  	val port = 8888
     val captcha = new Captcha
-    val result = captcha.getCaptcha()
-    println(result)
-    captcha.display()
-    captcha.closeConnection()
+    val server = new HTTPServer(port)
+    val host = server.getVirtualHost(null)
+
+    implicit val formats = DefaultFormats
+    
+    host.addContext("/v1/captcha",(req, resp) => {
+    	val body = req.getJson()
+    	val json = parse(body)
+    	val param = json.extract[Parameters]
+    	val id = captcha.getChallenge(param)
+    	resp.getHeaders().add("Content-Type","application/json")
+    	resp.send(200, write(id))
+    	0
+    })
+
+    host.addContext("/v1/media",(req, resp) => {
+    	val body = req.getJson()
+    	val json = parse(body)
+    	val id = json.extract[Id]
+    	val image = captcha.getCaptcha(id)
+    	resp.getHeaders().add("Content-Type","image/png")
+    	resp.send(200, image)
+    	0
+    })
+
+    host.addContext("/v1/answer",(req, resp) =>{
+    	val body = req.getJson()
+    	val json = parse(body)
+    	val answer = json.extract[Answer]
+    	val result = captcha.getAnswer(answer)
+    	resp.getHeaders().add("Content-Type","application/json")
+    	if(result){
+    		resp.send(200,"""{"result":"True"}""")
+    	}
+    	resp.send(200,"""{"result":"False"}""")
```

Comment:
Surround this with an `else` clause, so that the send is not executed twice.

Reply:
As a matter of good style, it would be even better to call `send` only once, and only choose the response in the `if..else` clause. For example:

```
val responseContent = if (result) """{"result":"True"}""" else """{"result":"False"}"""
resp.send(200, responseContent)
```

Reply:
The `resp.send` is executed only once in an end point. However, i will make the changes as you mentioned above.